infrastructure/merklemountainrange/src/merkle_change_tracker.rs

```diff
@@ -230,3 +228,101 @@ impl MerkleChangeTracker {
+    use blake2::Blake2b;
+    use digest::Digest;
+    use serde_derive::{Deserialize, Serialize};
+    use std::fs;
+    use tari_storage::lmdb::*;
+    use tari_utilities::Hashable;
+
+    #[derive(Serialize, Deserialize)]
+    pub struct IWrapper(u32);
+
+    impl Hashable for IWrapper {
+        fn hash(&self) -> Vec<u8> {
+            Blake2b::new().chain(self.0.to_le_bytes()).result().to_vec()
+        }
+    }
+
+    fn create_mmr(leaves: u32) -> MerkleMountainRange<IWrapper, Blake2b> {
+        let mut mmr: MerkleMountainRange<IWrapper, Blake2b> = MerkleMountainRange::new();
+        mmr.init_persistance_store(&"mmr".to_string(), 5);
+        for i in 1..leaves + 1 {
+            let object: IWrapper = IWrapper(i);
+            mmr.push(object);
+        }
+        mmr
+    }
+
+    #[test]
+    fn create_med_mmr() {
+        let _res = fs::remove_dir_all("./tests/test_mmr_cm"); // we ensure that the test dir is empty
+        let mut mmr = create_mmr(14);
+        // create storage
+        fs::create_dir("./tests/test_mmr_cm").unwrap();
+        let builder = LMDBBuilder::new();
+        let mut store = builder
+            .set_mapsize(5)
+            .set_path("./tests/test_mmr_cm/")
+            .add_database(&"mmr_mmr_checkpoints".to_string())
+            .add_database(&"mmr_mmr_objects".to_string())
+            .add_database(&"mmr_init".to_string())
+            .build()
+            .unwrap();
+        let result = mmr.apply_checkpoint(&mut store);
+        assert_eq!(result.is_ok(), true);
+        let mut mmr2: MerkleMountainRange<IWrapper, Blake2b> = MerkleMountainRange::new();
+        mmr2.init_persistance_store(&"mmr".to_string(), 5);
+        let result = mmr2.load_from_store(&mut store);
+        assert_eq!(result.is_ok(), true);
+        assert_eq!(mmr.get_merkle_root(), mmr2.get_merkle_root());
+
+        // add more leaves
+        for i in 15..25 {
+            let object: IWrapper = IWrapper(i);
+            mmr.push(object);
+            assert_eq!(mmr.change_tracker.objects_to_save.len() > 0, true);
+            let result = mmr.apply_checkpoint(&mut store);
+            assert_eq!(mmr.change_tracker.objects_to_save.len() == 0, true);
+            assert_eq!(result.is_ok(), true);
+            let mut mmr2: MerkleMountainRange<IWrapper, Blake2b> = MerkleMountainRange::new();
+            mmr2.init_persistance_store(&"mmr".to_string(), 5);
+            let result = mmr2.load_from_store(&mut store);
+            assert_eq!(result.is_ok(), true);
+            assert_eq!(mmr.get_merkle_root(), mmr2.get_merkle_root());
+        }
+
+        // add much more leafs
+        for i in 26..50 {
+            let object: IWrapper = IWrapper(i);
+            mmr.push(object);
+            let object_delete = IWrapper(i - 25);
+            let result = mmr.prune_object_hash(&object_delete.hash());
+            assert_eq!(result.is_ok(), true);
+            assert_eq!(mmr.change_tracker.objects_to_save.len() > 0, true);
```

Comment:
This can also be applied in a few places:
```suggestion
            assert!(mmr.change_tracker.objects_to_save.len() > 0);
```